src/server.rs

```diff
@@ -1,21 +1,85 @@
 //! Faucet server implementation.
 
+use std::collections::HashSet;
+use std::net::{IpAddr, ToSocketAddrs as _};
+use std::str::FromStr as _;
+use std::sync::{Arc, RwLock};
+use std::time::{Duration, SystemTime};
+
 use actix_cors::Cors;
 use actix_web::http::{header, StatusCode};
-use actix_web::web::{get, post, Bytes};
-use actix_web::{App, HttpResponse, HttpServer, Responder};
-use eyre::Result;
+use actix_web::web::{get, post, Bytes, Data};
+use actix_web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use eyre::{eyre, Result};
 use tracing::{error, info};
 
-use crate::{active_requests, config, erc20_tokens, id, neon_token};
+use crate::{active_requests, config, erc20_tokens, id, neon_token, solana};
+
+type AirdropLimiter = Data<RwLock<neon_token::AirdropLimiter>>;
 
 /// Starts the server in listening mode.
 pub async fn start(workers: usize) -> Result<()> {
     let rpc_bind = config::rpc_bind();
     let rpc_port = config::rpc_port();
     info!("{} Bind {}:{}", id::default(), rpc_bind, rpc_port);
 
-    HttpServer::new(|| {
+    let blacklist = config::blacklisted_ips()
+        .into_iter()
+        .map(|ip| IpAddr::from_str(&ip).map_err(|err| eyre!("Invalid blacklisted ip: {}", err)))
+        .collect::<Result<_>>()?;
+    let mut trusted_proxies = HashSet::new();
```

Comment:
Will there be some way to still run command-line faucet requests for local testing? Maybe if the requesting IP is localhost?

Reply:
are you asking for unlimited requests? In local testing just up the cap?

Reply:
I think I misunderstood how the proxy logic worked; resolving this comment.